src/server/game/Entities/Player/Player.cpp

```diff
@@ -2665,6 +2667,31 @@ void Player::GiveLevel(uint8 level)
     sScriptMgr->OnPlayerLevelChanged(this, oldLevel);
 }
 
+uint8 Player::GetExpansionLevelLimit() const
+{
+    uint8 max_lvl = GetMaxLevelForExpansion(_acc_expansion);
+
+    // if _acc_expansion stores some undefined value, GetMaxLevelForExpansion
+    // returns 0 and it should be treated like the latest expansion
+    // with deafult max lvl
+    return max_lvl != 0 ? max_lvl : DEFAULT_MAX_LEVEL;
```

Comment:
Don't do weird corrections of returned value here, change GetMaxLevelForExpansion to return DEFAULT_MAX_LEVEL instead (and remove this function)

Reply:
good catch